sound/pci/hda/patch_hdmi.c

```diff
@@ -2654,7 +2654,7 @@ static void generic_acomp_pin_eld_notify(void *audio_ptr, int port, int dev_id)
 	/* skip notification during system suspend (but not in runtime PM);
 	 * the state will be updated at resume
 	 */
-	if (snd_power_get_state(codec->card) != SNDRV_CTL_POWER_D0)
+	if (snd_power_get_state(codec->card) >= SNDRV_CTL_POWER_D3)
```

Comment:
for my understanding, what does `snd_power_get_state()` actually return here?

Reply:
@lyakh It will return SNDRV_CTL_POWER_D2, which is set in soc-core.c. The latest patch takes a bit different approach.